packages/cmrjs/cmr-utils.js

```diff
@@ -54,6 +54,22 @@ function isCMRFile(fileobject) {
   return isCMRFilename(cmrfilename);
 }
 
+/**
+ * Reduce granule object array to CMR files array
+ *
+ * @param {Object} granules = granule objects array
+ *
+ * @returns {Array<Object>} - CMR file object array: { filename, granuleId }
+ */
+function reduceGranulesToCmrFileObjects(granules) {
+  const reducer = (cFiles, g) => cFiles.concat(
+    g.files.filter(isCMRFile).map(
+      (cf) => ({ filename: cf.filename, granuleId: g.granuleId })
+    )
+  );
+  return granules.reduce(reducer, []);
```

Comment:
Nitpicky, feel free to ignore, but this might be a little easier to read with `lodash.flatten`.  `return flatten(granules.map(...));`

Reply:
The reason I'm using a reducer is because not all/any granules may necessarily have CMR files. It doesn't seem like lodash.flatten really helps with that? Correct me if I'm wrong.

Reply:
```js
const granuleToCmrFileObjects = (granule) =>
  granule.files
    .filter(isCMRFile)
    .map((f) => ({ filename: f.filename, granuleId: g.granuleId }));

const granulesToCmrFileObjects = (granules) =>
  flatten(granules.map(granuleToCmrFileObjects));
```

Reply:
Ah, right. It wasn't clear to me from the docs whether `flatten` drops empty results. Looks like it does.